File: Cpp/53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int x = nums[0];
        int sol = INT_MIN;
        for(int i = 1; i < nums.size(); i++){
            sol = max(sol, x);
            x = max(nums[i], x + nums[i]);
        }
        sol = max(sol, x);
        return sol;
    }
};

File: Python/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sol = nums[0]
        curr = nums[0]
        if len(nums) == 1:
            return sol
        for i in range(1, len(nums)):
            if curr + nums[i] < nums[i]:
                curr = nums[i]
            else:
                curr += nums[i]
            sol = max(sol, curr)
        return sol
